common/distribution/dist.go

```diff
@@ -0,0 +1,77 @@
+// Copyright 2016 The rkt Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package distribution
+
+import (
+	"fmt"
+	"net/url"
+)
+
+// DistType represents a distribution type
+type DistType string
+
+const (
+	DistScheme = "cimd"
+
+	// Distribution types
+	DistTypeAppc       DistType = "appc"
+	DistTypeACIArchive DistType = "aci-archive"
+	DistTypeDocker     DistType = "docker"
+)
+
+// A Distribution represnt the way to retrieve an image starting from an input
+// string.
+// It's identified by an URI with a specific schema
+type Distribution interface {
+	// Returns a copy of the distribution URI
+	URI() *url.URL
+	// Returns the distribution type
+	Type() DistType
```

Comment:
>  One question regarding the "distribution handler": Couldn't this also be an interface type included in the returned struct from a potential Parse(...) ... method? Just fighting for more generalization ;-)

The distribution handlers, like the user friendly string and the use of distribution URIs as references in #3071 are on top of this. Every user could implement their own logic. Given this, I'd like that this could be shared be more projects since it could probably be of help also for k8s (@euank @yifan-gu) for handling multiple diatribution types (sometimes called, IMHO not correctly, image types).